### PR TITLE
Article/seeds of inception 5

### PR DESCRIPTION
**Pull request by mechanicalpete**

### CODE REVIEW

File: _posts/2018-07-18-inception-pipelines-pt5.markdown [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+|root/aws_capability_iam.yml | Defines the ChildDeveloperPolicy and ChildDeveloperGroup.|
+|root/aws_capability_iam_users.yml | Defines the IAM User (in this case DeveloperPete) and assigns them to the ChildDeveloperGroup|
+
+## Taking It For A Spin
+
+If you're new to Inception Pipelines, head on over to the original post and set yourself up a pipeline. If not, copy over the files referenced above, add in the appropriate pipeline action and commit. Enjoy!
+
+## Wrapping Up
+
+And so, another month closes with another awesome Inception Pipeline post.
```
Comment (jgunnink) [inline _posts/2018-07-18-inception-pipelines-pt5.markdown]: Whilst true, you don't actually wrap up the post into a bite-sized conclusion or... wrap-up if you will

File: _posts/2018-07-18-inception-pipelines-pt5.markdown [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+|File|Description|
+|----|-----------|
+|child/aws_capability_iam.yml | Defines the DeveloperPolicy and DeveloperRole. DeveloperRole is the role that is assumed by your developers from the root account.|
+|root/aws_capability_iam.yml | Defines the ChildDeveloperPolicy and ChildDeveloperGroup.|
+|root/aws_capability_iam_users.yml | Defines the IAM User (in this case DeveloperPete) and assigns them to the ChildDeveloperGroup|
+
+## Taking It For A Spin
+
+If you're new to Inception Pipelines, head on over to the original post and set yourself up a pipeline. If not, copy over the files referenced above, add in the appropriate pipeline action and commit. Enjoy!
```
Comment (jgunnink) [inline _posts/2018-07-18-inception-pipelines-pt5.markdown]: Don't feel like this section adds any value?

File: _posts/2018-07-18-inception-pipelines-pt5.markdown [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+region         = ap-southeast-2
+source_profile = pete
+role_arn       = arn:aws:iam::222233334444:role/DeveloperRole
+```
+
+If you had multiple roles, then copy-and-paste the above and give it another name.
+
+## Where Do I Get The Seed Files
+
+Two complete, working pipelines are available [here on GitHub](https://github.com/MechanicalRock/InceptionPipeline/tree/post/part-5) with one pipeline for your root account and one for your child account. As discussed in the [last post](https://mechanicalrock.github.io//aws/continuous/deployment/codepipeline/codebuild/inception/pipeline/2018/06/25/inception-pipelines-pt4), these would be considered account-level pipelines.
```
[the remark this answers is not in the document]
Comment (mechanicalpete) [reply _posts/2018-07-18-inception-pipelines-pt5.markdown]: I'll link part of the text in the first paragraph to the previous post.

File: _posts/2018-07-18-inception-pipelines-pt5.markdown [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+**Part 5 access all accounts** - A practical example for DevOps on AWS
+
+![vines]({{ site.url }}/img/inception-pipelines/part-5-vines.jpg)
+
+## What's The Problem
+
+This month we're building on last month's post and discussing one approach to managing IAM users across multiple accounts. More specifically we're avoiding the quick, dirty and lazy way of just throwing IAM users into each account. When you do, you quickly end up with an unmanageable jungle of profiles and access keys, which creates an increased exposure to compromised accounts. So here is a better way; read on fair explorers while we cut a path through this jungle to IAM user nirvana.
+
+It is worth explicitly calling out that the ideas discussed below are more targeted towards smaller AWS installations. If you're working in a large corporate environment with a dedicated user directory and tens or hundreds of AWS Accounts, then there is a better way. [Reach out](https://www.mechanicalrock.io/#/contact-us) to us to discuss how a bespoke Cloud Native Platform can really transform your AWS usage!
```
[the remark this answers is not in the document]
Comment (mechanicalpete) [reply _posts/2018-07-18-inception-pipelines-pt5.markdown]: Reworded this to remove the reference to CNP.

File: _posts/2018-07-31-inception-pipelines-pt5.markdown [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+image: img/inception-pipelines/seed_germination.png
+---
+
+**Part 5 access all accounts** - A practical example for DevOps on AWS
+
+![vines]({{ site.url }}/img/inception-pipelines/part-5-vines.jpg)
+
+## What's The Problem
+
+This month we're building on [last month's post](https://mechanicalrock.github.io//aws/continuous/deployment/codepipeline/codebuild/inception/pipeline/2018/06/25/inception-pipelines-pt4) and discussing one approach to managing IAM users across multiple accounts. More specifically we're avoiding the quick, dirty and lazy way of just throwing IAM users into each account. When you do, you quickly end up with an unmanageable jungle of profiles and access keys, which creates an increased exposure to compromised accounts. So here is a better way; read on fair explorers while we cut a path through this jungle to IAM user nirvana.
```
Comment (jgunnink) [inline _posts/2018-07-31-inception-pipelines-pt5.markdown]: Nice lead-in 👍

**Comment by temyers** [comment]
@mechanicalpete - can you fix the Snyk issues?

File: _posts/2018-07-31-inception-pipelines-pt5.markdown [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+        Password: Dummy12321ymmuD
+        PasswordResetRequired: true
+      UserName: Pete
+
+  AddUsersToChildDeveloperGroup:
+    Type: AWS::IAM::UserToGroupAddition
+    Properties:
+      GroupName: !ImportValue 'RootAccountCapabilityIamChildDeveloperGroupName'
+      Users:
+        - !Ref DeveloperPete
```
Comment (jgunnink) [inline _posts/2018-07-31-inception-pipelines-pt5.markdown]: You misspelt PipelinePete

File: _posts/2018-07-31-inception-pipelines-pt5.markdown [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+      Groups:
+        - !Ref ChildDeveloperGroup
+      PolicyDocument:
+        Statement:
+          - Action: sts:AssumeRole
+            Effect: Allow
+            Resource: !Sub 'arn:aws:iam::${ParamChildAccount}:role/${ParamChildAccountRoleName}'
+```
+
+The final file is `root/aws_capability_iam_users.yml`. In this file you create your IAM User and then assign them to the group(s) that they need access to. If the developer leaves the company, simply delete the IAM User and any group assignments and redeploy. Within a minute they will no longer have any access. Alternatively, if it's a temporary change such as long service leave, you can just remove them from the group(s) and re-add them on return.
```
Comment (jgunnink) [inline _posts/2018-07-31-inception-pipelines-pt5.markdown]: Nice. Especially good examples 👍

**Pull request merged**